src/StatusQ/Buttons/StatusButton.qml

```diff
@@ -0,0 +1,165 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+
+Rectangle {
+    id: statusButton
+
+    enum Size {
+        Small,
+        Large
+    }
+
+    enum Type {
+        Normal,
+        Danger,
+        WithoutBackground,
+        DangerWithoutBackground
+    }
+
+    property bool loading: false
+
+    property int size: StatusButton.Size.Large
+    property int type: StatusButton.Type.Normal
+
+    property alias text: label.text
+    property alias font: label.font
+
+    property real leftPadding: size === StatusButton.Size.Large ? 24 : 12 // by design
+    property real rightPadding: size === StatusButton.Size.Large ? 24 : 12 // by design
+    property real topPadding: size === StatusButton.Size.Large ? 11 : 10 // by design
+    property real bottomPadding: size === StatusButton.Size.Large ? 11 : 10 // by design
+
+    signal pressed(var mouse)
+
+
+
+    /// Implementation
+
+    implicitWidth: sensor.width
+    implicitHeight: sensor.height
+
+    radius: 8 // by design
+    color: {
+        switch(statusButton.type) {
+        case StatusButton.Type.Normal:
+            if (statusButton.enabled) {
+                return sensor.containsMouse ? Theme.palette.primaryColor2
+                                            : Theme.palette.primaryColor3;
+            } else {
+                return Theme.palette.baseColor2;
+            }
```

Comment:
Tiny nitpick: since we're return in any case if `statusButton.enabled`, we can have the second `return` without the `else` branch